docs/src/lints/high_cyclomatic_complexity.md

```diff
@@ -0,0 +1,23 @@
+# high_cyclomatic_complexity
+## What it does
+Measures the cyclomatic complexity of a function to see if it exceeds the configure maximum.
+
+## Why this is bad
+High branch complexity can lead to functions that are hard to test, and harder to reason about.
+
+## Example
+```lua
+return function(props)
```

Comment:
This is a pretty hard to understand example for a documentation. Any chance we can get something more helpful like https://eslint.org/docs/latest/rules/complexity?

Fun fact: Clippy's example is just "You'll know it when you see it" :D